lib/components/segmented-control/base.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import clsx from "clsx";
+import { TGenericSizes } from "../../types";
+import { Segment } from "./segment";
+import "./segmented-control.scss";
+
+export interface SegmentedControlProps {
+    className?: string;
+    options: Array<{
+        disabled?: boolean;
+        icon?: string | React.ReactNode;
+        label?: string;
+        selected?: boolean;
+    }>;
+    onChange?: (selectedItemIndex: number) => void;
+    hasContainerWidth?: boolean;
+    size?: Extract<TGenericSizes, "lg" | "md" | "sm">;
```

Comment:
I have added `TRegularSizes` in the generic types, we can use that instead. Thanks

Reply:
thank you so much :)